### PR TITLE
Feature/12 remove welcome message for the endpoint with form

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] Remove "welcome message" for the endpoint with the form
- [x] Update unit test: twilio_endpoint

### CODE REVIEW

**Review by dedenbangkit — CHANGES_REQUESTED** [review]
Please write test

**Comment by dedenbangkit** [comment]
You can write same test: [#L42-L250](https://github.com/akvo/Akvo-DjangoFormGateway/blob/df295737e1964e98def8758c1f9379e7bbf76c02/src/AkvoDjangoFormGateway/tests/tests_twilio_endpoint.py#L42-L250) (with some modification)

Also give details about the response text in the line [#47](https://github.com/akvo/Akvo-DjangoFormGateway/blob/df295737e1964e98def8758c1f9379e7bbf76c02/src/AkvoDjangoFormGateway/tests/tests_twilio_endpoint.py#L47)

**Comment by ifirmawan** [comment]
> #47

noted, on it

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        self.assertEqual(response.json(), f"{question.order}. {question.text}")
+
+        # Answer wrong photo question no MediaContentType0
+        image = "http://twilio.example/image/caseSensiT1Ve2.png"
+        json_form = {
+            "Body": "",
+            "From": f"whatsapp:+{phone_number}",
+            "MediaUrl0": image,
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:76]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+            },
+        )
+
+        image_type = "image/png"
+        json_form = {
+            "Body": "",
+            "From": f"whatsapp:+{phone_number}",
+            "MediaContentType0": image_type,
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:95]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+        )
+
+        # Answer right photo question
+        json_form = {
+            "Body": "",
+            "From": f"whatsapp:+{phone_number}",
+            "MediaContentType0": image_type,
+            "MediaUrl0": image,
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:115]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+            "Body": "",
+            "From": f"whatsapp:+{phone_number}",
+            "Longitude": lng,
+        }
+        text = feed.get_answer_text(
+            body=None,
+            image_url=None,
+            lng=lng,
+        )
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:159]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+            "Latitude": lat,
+            "Longitude": lng,
+        }
+        text = feed.get_answer_text(
+            body=None,
+            image_url=None,
+            lat=lat,
+            lng=lng,
+        )
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:175]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+
+        # Phone question
+        question = feed.get_question(form=datapoint.form, data=datapoint)
+        # Answer Phone question
+        reply_text = "62819111"
+        json_form = {
+            "Body": reply_text,
+            "From": f"whatsapp:+{phone_number}",
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:194]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+
+        # Single option question
+        question = feed.get_question(form=datapoint.form, data=datapoint)
+        # Answer Single option question
+        reply_text = "OptiOn 2"
+        json_form = {
+            "Body": reply_text,
+            "From": f"whatsapp:+{phone_number}",
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:213]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+
+        # Multiple option question
+        question = feed.get_question(form=datapoint.form, data=datapoint)
+        # Answer Multiple option question
+        reply_text = "multi 1, MULTI 3"
+        json_form = {
+            "Body": reply_text,
+            "From": f"whatsapp:+{phone_number}",
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:232]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+        # Date question
+        question = feed.get_question(form=datapoint.form, data=datapoint)
+
+        # Answer wrong date question
+        reply_text = "15/12/1999"
+        json_form = {
+            "Body": reply_text,
+            "From": f"whatsapp:+{phone_number}",
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:256]: Wrong url. it should be `/api/gateway/twilio/1`

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+                text=reply_text, question=question, data=datapoint
+            )
+        )
+        # Answer date question
+        reply_text = "15-12-1999"
+        json_form = {
+            "Body": reply_text,
+            "From": f"whatsapp:+{phone_number}",
+        }
+        response = client.post("/api/gateway/twilio/", json_form)
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py:269]: Wrong url. it should be `/api/gateway/twilio/1`

**Review by dedenbangkit — CHANGES_REQUESTED** [review]
Please understand the context @ifirmawan

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 11]
```diff
@@ -40,244 +36,34 @@ def test_instance_not_found(self):
...
-        self.assertEqual(
-            response.json(),
-            {
-                "non_field_errors": [
-                    "MediaUrl0 is required when MediaContentType0 is present."
-                ]
-            },
-        )
-
-        # Answer right photo question
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py]: restore this line

File: src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py [hunk 12]
```diff
@@ -40,244 +36,34 @@ def test_instance_not_found(self):
...
-        }
-        response = client.post("/api/gateway/twilio/", json_form)
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertFalse(
-            feed.validate_answer(
-                text=reply_text, question=question, data=datapoint
-            )
-        )
-        # Answer date question
-        reply_text = "15-12-1999"
```
Comment (dedenbangkit) [inline src/AkvoDjangoFormGateway/tests/tests_twilio_instance_endpoint.py]: restore this line

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**